src/background/zcl_abapgit_background.clas.abap

```diff
@@ -59,36 +59,75 @@ CLASS zcl_abapgit_background IMPLEMENTATION.
 
   METHOD list_methods.
 
-    DATA: ls_method       LIKE LINE OF rt_methods,
-          ls_key          TYPE seoclskey,
-          lt_implementing TYPE seor_implementing_keys,
-          ls_implementing LIKE LINE OF lt_implementing.
-
-    FIELD-SYMBOLS: <ls_method> LIKE LINE OF rt_methods.
-
-
-* in order to handle local classes in the compiled report
-    ls_method-class = 'ZCL_ABAPGIT_BACKGROUND_PULL'.
-    INSERT ls_method INTO TABLE rt_methods.
-    ls_method-class = 'ZCL_ABAPGIT_BACKGROUND_PUSH_AU'.
-    INSERT ls_method INTO TABLE rt_methods.
-    ls_method-class = 'ZCL_ABAPGIT_BACKGROUND_PUSH_FI'.
-    INSERT ls_method INTO TABLE rt_methods.
-
-    ls_key-clsname = 'ZIF_ABAPGIT_BACKGROUND'.
-
-    CALL FUNCTION 'SEO_INTERFACE_IMPLEM_GET_ALL'
-      EXPORTING
-        intkey       = ls_key
-      IMPORTING
-        impkeys      = lt_implementing
-      EXCEPTIONS
-        not_existing = 1
-        OTHERS       = 2 ##FM_SUBRC_OK.
-    LOOP AT lt_implementing INTO ls_implementing.
-      ls_method-class = ls_implementing-clsname.
-      INSERT ls_method INTO TABLE rt_methods.
-    ENDLOOP.
+    CONSTANTS: BEGIN OF c_name,
+                 interface TYPE seoclskey VALUE 'ZIF_ABAPGIT_BACKGROUND',
+                 report    TYPE syrepid VALUE 'ZABAPGIT_STANDALONE',
+               END OF c_name.
+
+    DATA: ls_method          LIKE LINE OF rt_methods,
+          lt_implementing    TYPE seor_implementing_keys,
+          ls_implementing    LIKE LINE OF lt_implementing,
+          lt_interf          TYPE abap_intfdescr_tab,
+          lt_local_classes   TYPE STANDARD TABLE OF scompo,
+          lv_classname       TYPE string,
+          lr_typedescr       TYPE REF TO cl_abap_typedescr,
+          lr_typedescr_class TYPE REF TO cl_abap_classdescr.
+
+    FIELD-SYMBOLS: <ls_local_class> LIKE LINE OF lt_local_classes,
+                   <ls_method>      LIKE LINE OF rt_methods.
+
+
+    IF sy-repid = c_name-report.
```

Comment:
You can copy the standalone report for example to keep different versions.

Use this instead:

```abap
IF zcl_abapgit_factory=>get_environment( )->is_merged( ) = abap_true.
```